migrations/3_deploy_adminwallet.js

```diff
@@ -0,0 +1,58 @@
+const AdminWallet = artifacts.require("./AdminWallet.sol");
+
+// if (process.env.NODE_ENV !== 'production') { // https://codeburst.io/process-env-what-it-is-and-why-when-how-to-use-it-effectively-505d0b2831e7
+require("dotenv").load();
+// }
+
+const PrivateKeyProvider = require("truffle-hdwallet-provider-privkey");
+const HDWalletProvider = require("truffle-hdwallet-provider");
+
+const admin_mnemonic = process.env.ADMIN_MNEMONIC;
+const infura_api = process.env.INFURA_API;
+
+module.exports = async function(deployer, network, provider) {
+
+	if( network != 'ganache' && network != 'test'
+		&& network != 'coverage') {
+
+		const adminWallet = await AdminWallet.deployed();
+		const networkSettings = settings[network] || settings["default"];
+		
+		let adminProvider;
+
+		await web3.eth.getAccounts(function(err, res) {
```

Comment:
accounts = await web3.eth.getAccounts()

Reply:
why do you need it? you should get the default accounts from the migration function parameter, and it (web3) should default the the 1st account anyways

Reply:
I removed this